.github/workflows/bundle-size.yml

```diff
@@ -0,0 +1,79 @@
+name: Bundle Size Analysis
+
+on:
+  pull_request:
+    paths:
+      - "packages/core/src/components/**"
+      - "packages/core/package.json"
+
+permissions:
+  pull-requests: write
+
+jobs:
+  build:
+    name: Build
+    uses: ./.github/workflows/build-and-upload.yml
+    secrets:
+      npm_token: ${{ secrets.npm_token }}
+
+  bundle-size:
+    runs-on: ubuntu-latest
+    needs: build
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: "yarn"
+
+      - name: Run Setup
+        uses: ./.github/actions/setup
+
+      - name: Download build artifacts
+        uses: ./.github/actions/download-builds
+
+      - name: Run size-limit on PR
+        run: |
+          yarn install --frozen-lockfile
+          node scripts/bundle-check/generate-size-limit-config.js
+          mkdir -p scripts/bundle-check/reports
+          ./node_modules/.bin/size-limit --json > "scripts/bundle-check/reports/pr.json"
+
+      - name: Run size-limit on base commit
+        run: |
+          PR_HEAD_SHA=$(git rev-parse HEAD)
+          trap "echo 'Checking out back to PR commit'; git checkout --force $PR_HEAD_SHA" EXIT
+
+          BASE=$(git merge-base origin/master HEAD)
+          echo "Base commit: $BASE"
+          git checkout $BASE
+          yarn install --frozen-lockfile
+          yarn build
+          yarn add --dev -W size-limit @size-limit/preset-small-lib
+          ./node_modules/.bin/size-limit --json > scripts/bundle-check/reports/base.json
+        continue-on-error: true
```

Comment:
I believe this entire step can be simplified using GitHub's Context, and separate the checkout to another step, but if you tried and didn't work let's check that out later